### PR TITLE
Implement Button IO functions

### PR DESCRIPTION
**Pull request by fustilio**
Define three button states: Not pressed, short press and long press

Implement Button A/B IO via interrupts, function returns either short press or not pressed 
Implement Power button via interrupts, function returns either not pressed, short or long press.

IOButtons are instantiated in TS_UI::begin()

Edit: Modified power button to use interrupts as well, requires setIntActiveLow method created in PR https://github.com/bettr-xyz/M5StickC/pull/4

### CODE REVIEW

File: alpha/hal.h [hunk 1]
```diff
@@ -104,6 +111,7 @@ class _TS_HAL
     //
     void led_set(TS_Led, bool);
     bool btn_a_get();
+    TS_ButtonState btn_power_get();
```
Comment (jamesi8086) [inline alpha/hal.h:110]: could you add in btn_b as well for completeness? should be pretty trivial

File: alpha/ui.cpp [hunk 2]
```diff
@@ -19,6 +20,11 @@ void _TS_UI::staticTask(void* parameter)
 
 void _TS_UI::begin()
 {
+
+  this->buttonA = new IOButton(BUTTONA);
```
Comment (jamesi8086) [inline alpha/ui.cpp]: can all the IOButton things be in/part of HAL? all IOs should be in or beneath HAL

**Comment by jamesi8086** [comment]
the button IO things look good, integrate it with HAL

File: alpha/hal.h (hunk 1 above)
Comment (fustilio) [reply alpha/hal.h:110]: yup sure I'll make the changes together with moving stuff under HAL

File: alpha/ui.cpp (hunk 2 above)
Comment (fustilio) [reply alpha/ui.cpp]: Yup sure I'll move this instantiating to HAL

File: alpha/io.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#define __TS_IO__
+
+enum TS_ButtonState
+{
+  NotPressed,
+  Short,
+  Long,
+};
+
+class IOButton
```
Comment (jamesi8086) [inline alpha/io.h]: can we prefix with TS_ ?
helps to reduce namespace collision with other libraries

Comment (jamesi8086) [reply alpha/io.h]: iobutton.h
class TS_IOButton

File: alpha/hal.cpp [hunk 4]
```diff
@@ -214,11 +221,57 @@ void _TS_HAL::led_set(TS_Led led, bool enable)
 #endif
 }
 
-bool _TS_HAL::btn_a_get()
+TS_ButtonState handleGPIOButton()
```
Comment (jamesi8086) [inline alpha/hal.cpp]: function name style please

handle_gpio_button
or
button_handle_gpio

[category/object][purpose]

File: alpha/hal.cpp [hunk 5]
```diff
@@ -214,11 +221,57 @@ void _TS_HAL::led_set(TS_Led led, bool enable)
...
 {
-#ifdef HAL_M5STICK_C
-  return M5.BtnA.read() == 1;
-#endif
+  return TS_ButtonState::Short;
+}
+
+TS_ButtonState handlePowerButton()
```
Comment (jamesi8086) [inline alpha/hal.cpp]: button_handle_power

**Review by jamesi8086 — APPROVED** [review]
a few more changes then it looks good to merge

**Pull request merged**